codex-rs/core/src/model_provider_info.rs

```diff
@@ -161,6 +172,23 @@ impl ModelProviderInfo {
             None => Ok(None),
         }
     }
+
+    /// Effective maximum number of request retries for this provider.
+    pub fn request_max_retries(&self) -> u64 {
+        self.request_max_retries.unwrap_or(4)
```

Comment:
I think the default values should be their own constants at the top of the file.